sympy/solvers/diophantine/diophantine.py

```diff
@@ -51,6 +52,86 @@
     "univariate"}
 
 
+class DiophantineSolutionSet(set):
+    """
+    Container for a set of solutions to a particular diophantine equation.
+
+    The base representation is a set of tuples representing each of the solutions.
+
+    Parameters
+    ==========
+
+    symbols : list
+        List of free symbols in the original equation.
+    parameters: list (optional)
+        List of parameters to be used in the solution
+
+    Examples
+    ========
+
+    Adding solutions:
+
+        >>> from sympy.solvers.diophantine.diophantine import DiophantineSolutionSet
+        >>> from sympy.abc import x, y, t, u
+        >>> s1 = DiophantineSolutionSet([x, y], [t, u])
+        >>> s1
+        set()
+        >>> s1.add(2, 3)
+        >>> s1.add(-1, u)
+        >>> s1
+        {(-1, u), (2, 3)}
+        >>> s2 = DiophantineSolutionSet([x, y], [t, u])
+        >>> s2.add(3, 4)
+        >>> s1.update(s2)
+        >>> s1
+        {(-1, u), (2, 3), (3, 4)}
+
+    Conversion of solutions into dicts:
+
+        >>> list(s1.dict_iterator())
+        [{x: -1, y: u}, {x: 2, y: 3}, {x: 3, y: 4}]
+
+    Substituting values:
+
+        >>> s3 = DiophantineSolutionSet([x, y], [t, u])
+        >>> s3.add(t**2, t + u)
+        >>> s3
+        {(t**2, t + u)}
+        >>> s3.substitute_parameters(2, 3)
+        {(4, 5)}
+        >>> s3.substitute_parameters(-1)
+        {(1, u - 1)}
+    """
+
+    def __init__(self, symbols_list, parameters=None):
+        super().__init__()
+        self.symbols = symbols_list
+
+        if parameters is None:
+            self.parameters = symbols('%s1:%i' % ('t', len(self.symbols) + 1), integer=True)
+        else:
+            self.parameters = parameters
+
+    def add(self, *solution):
```

Comment:
I tend to think that it is good not to use `*args` in cases like this. It is better to require the user to supply a tuple. In this case it would also be consistent with the interface of `set`.